packages/@webex/plugin-cc/src/services/core/WebSocket/connection-service.ts

```diff
@@ -32,20 +22,22 @@ export class ConnectionService extends EventTarget {
   private isKeepAlive: boolean;
   private reconnectInterval: ReturnType<typeof setInterval>;
   private webSocketManager: WebSocketManager;
-  private subscribeRequest: SubscribeRequest;
+  private eventBus: EventBus;
+  private onReRegister: () => Promise<void>;
 
-  constructor(webSocketManager: WebSocketManager, subscribeRequest: SubscribeRequest) {
+  constructor(options: ConnectionServiceOptions) {
     super();
-    this.webSocketManager = webSocketManager;
-    this.subscribeRequest = subscribeRequest;
+    this.webSocketManager = options.webSocketManager;
+    this.onReRegister = options.onReRegister;
```

Comment:
We register the callback here, it will be called when we reconnect to the socket again.